golang/cosmos/x/vtransfer/keeper/keeper.go

```diff
@@ -101,12 +102,42 @@ func (k Keeper) GetReceiverImpl() vibctypes.ReceiverImpl {
 	return k
 }
 
+// Extract the base address from the packet sender (if senderIsLocal) or
+// receiver (if !senderIsLocal), since the local ibcModule doesn't understand
+// address parameters.
+func (k Keeper) packetWithOnlyBaseAddresses(packet channeltypes.Packet, senderIsLocal bool) channeltypes.Packet {
+	transferData := transfertypes.FungibleTokenPacketData{}
+	if err := k.cdc.UnmarshalJSON(packet.GetData(), &transferData); err != nil {
+		return packet
+	}
+	if senderIsLocal {
+		baseSender, err := types.ExtractBaseAddress(transferData.Sender)
+		if err == nil {
+			transferData.Sender = baseSender
+		}
+	} else {
+		baseReceiver, err := types.ExtractBaseAddress(transferData.Receiver)
+		if err == nil {
+			transferData.Receiver = baseReceiver
+		}
+	}
+
+	// Re-serialize the packet data with the base addresses.
+	data, _ := k.cdc.MarshalJSON(&transferData)
+
+	// Replace the packet's data with the new data.
+	packet.Data = data
+	return packet
+}
```

Comment:
I don't like that this function mutates its argument _and_ returns it, which can lead to downstream confusion—or that the `senderIsLocal` argument is a [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap). Suggestion:
```suggestion
// stripAddressDecoration removes all subpath and query components from either
// the Sender or Receiver address in a packet's data so it can be consumed by a
// local ibcModule that understands only base addresses.
func (k Keeper) stripAddressDecoration(packet channeltypes.Packet, role AddressRole) error {
	// Extract.
	packetData := transfertypes.FungibleTokenPacketData{}
	if err := k.cdc.UnmarshalJSON(packet.GetData(), &packetData); err != nil {
		return err
	}

	// Transform.
	switch role {
	case RoleSender:
		baseSender, err := types.ExtractBaseAddress(packetData.Sender)
		if err != nil {
			return err
		}
		packetData.Sender = baseSender
	case RoleReceiver:
		baseReceiver, err := types.ExtractBaseAddress(packetData.Receiver)
		if err != nil {
			return err
		}
		packetData.Receiver = baseReceiver
	default:
		return fmt.Errorf("invalid role: %v", role)
	}

	// Replace.
	remarshalled, err := k.cdc.MarshalJSON(&packetData)
	if err != nil {
		return err
	}
	packet.Data = remarshalled

	return nil
}
```
with
```go
type AddressRole string

const (
	AddressRole RoleSender   = "Sender"
	AddressRole RoleReceiver = "Receiver"
)
```